apps/versions/tests.py

```diff
@@ -833,6 +840,22 @@ def test_success(self):
         assert self.addon.current_version
         self.assert_served_by_mirror(self.client.get(self.latest_url))
 
+    def test_beta(self):
+        r = self.client.get(self.latest_beta_url)
+        eq_(r.status_code, 302)
+        beta_file_url = reverse('downloads.file', args=[self.beta_file.id])
+        url = beta_file_url + '/' + self.beta_file.filename
+        assert r['Location'].endswith(url), r['Location']
+
+        # No beta downloads for unreviewed addons
+        self.addon.status = amo.STATUS_PENDING
+        self.addon.save()
+        r = self.client.get(self.latest_beta_url)
+        eq_(r.status_code, 302)
+        beta_file_url = reverse('downloads.file', args=[self.beta_file.id])
+        url = self.file_url + '/' + self.file.filename
+        assert r['Location'].endswith(url), r['Location']
+
```

Comment:
Btw, I wouldn't mind if you split the test to have one small test for each of the three cases:
1/ public, with a beta version
2/ public, no beta version
3/ not public (with or without a beta version, we don't really care, we should never be in a state where a non public addon have a beta version, as beta uploads are only allowed on public addons)